scripts/benchmarks/run_formatter.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env sh
+
+###
+# Benchmark the Ruff formatter's performance against a variety of similar tools.
+#
+# Expects to be run from the repo root after invoking `cargo build --release`,
+# in an environment with access to `black`, `autopep8`, and `yapf` (most recently:
+# `black` v23.9.1, `autopep8` v2.0.4, and `yapf` v0.40.1).
+#
+# Example usage:
+#
+#   ./scripts/benchmarks/run_formatter.sh ~/workspace/zulip
+###
+
+TARGET_DIR=${1}
+
+# Benchmark 1: Write to disk.
+hyperfine --ignore-failure --warmup 1 --setup "cd ${TARGET_DIR} && git reset --hard HEAD" \
```

Comment:
I'm not entirely sure but I believe `--setup` only runs once. Meaning, subsequent runs don't work on a clean git directory. We may want to use `--prepare` instead

Reply:
This is intentional. We don't want to reset the directory on each invocation, otherwise the benchmarks will suffer from the issue you pointed out in the comment above.

Reply:
The example is to use `--setup` to compile your program. That means it only runs once for the whole benchmark. 

I guess the reset doesn't really matter because each formatter will format the file to their expected output as part of the warmup.

Reply:
So maybe it's worth adding a new benchmark that would at least partially test the formatting performance(with a note that probably most of the test files were previously reformatted according to black guidelines)

Reply:
Yeah, perhaps over a repo like CPython that (AFAIK) isn't formatted.